src/boilerplates/create_embedded_wallet_react.ts

```diff
@@ -14,52 +15,56 @@ export async function createEmbeddedWalletReact(program: Command, options: any)
     let googleClientId = options.googleClientId;
     let appleClientId = options.appleClientId;
 
-    if (!waasConfigKey) {
-        console.log("Please provide the WaaS Config Key for your project.");
-        console.log("Your config key can be found at https://sequence.build under the embedded wallet settings.");
-        console.log("To skip and use the default test config key, press enter.");
- 
-        waasConfigKey = await input({
-            message: "WaaS Config Key:",
-        });
-
-        console.log("");
-    }
-
-    if (!projectAccessKey && waasConfigKey != '') {
-        console.log("Please provide the Project Access Key for your project.");
-        console.log("Your access key can be found at https://sequence.build under the project settings.");
-        console.log("To skip and use the default test access key, press enter.");
- 
-        projectAccessKey = await input({
-            message: "Project Access Key:",
-        });
-
-        console.log("");
-    }
-
-    if (!googleClientId && waasConfigKey != '') {
-        console.log("Please provide the Google Client ID for your project.");
-        console.log("Your client ID can be found at https://console.cloud.google.com/apis/credentials");
-        console.log("To skip and use the default test client ID, press enter.");
- 
-        googleClientId = await input({
-            message: "Google Client ID:",
-        });
-
-        console.log("");
-    }
-
-    if (!appleClientId && waasConfigKey != '') {
-        console.log("Please provide the Apple Client ID for your project.");
-        console.log("Your client ID can be found at https://developer.apple.com/account/resources/identifiers/list/serviceId");
-        console.log("To skip and use the default test client ID, press enter.");
- 
-        appleClientId = await input({
-            message: "Apple Client ID:",
-        });
-
-        console.log("");
+    const userWantsToConfigureTheirKeys = await promptUserKeyCustomizationDecision();
+    
+    if (userWantsToConfigureTheirKeys) {
+        if (!waasConfigKey) {
```

Comment:
It feels like we are repeating these `waasConfigKey`, `projectAccessKey` checks and `await input` implementations all over the place. Can you centralize these into a single point as well? That way the implementation will be a lot cleaner.

Reply:
About that can you provide me some feedback about this? https://github.com/0xsequence/sequence-cli/pull/11/files#diff-0e4c4ef1bf4917d07d28ca11950aa794bab7500434ff881ba77fee6630501646. I want to know if this satisfies your idea and if it seems well structured to you.

Reply:
@RichardIrala Very good start. Now let's go one step further and do wrapper utility functions that use `promptForKeyWithLogs` under the hood, but specific to most common repeated cases. I suspect those will be `promptForWaaSConfigKeyWithLogs`, `promptForProjectAccessKeyWithLogs` etc.

Reply:
Done